src/app/vendors/VendorItems.tsx

```diff
@@ -14,27 +14,25 @@ import { VendorItem } from './vendor-item';
 import VendorItemComponent from './VendorItemComponent';
 import styles from './VendorItems.m.scss';
 
+function vendorItemIndex(item: VendorItem) {
+  return item.key;
+}
+
 function itemSort(category: string) {
   if (category === 'category.rank_rewards_seasonal') {
     return chainComparator<VendorItem>(
       compareBy((item) => item.item?.tier),
-      compareBy((item) => parseInt(item.item?.id ?? '', 10))
+      compareBy(vendorItemIndex)
     );
   } else if (category === 'category_bounties') {
     return chainComparator<VendorItem>(
       compareBy((item) => item.item?.typeName),
-      compareBy((item) => parseInt(item.item?.id ?? '', 10)),
-      compareBy((item) => item.item?.itemCategoryHashes)
+      compareBy(vendorItemIndex)
     );
-  } else if (category === 'category_weapon') {
-    return chainComparator<VendorItem>(compareBy((item) => item.item?.itemCategoryHashes));
```

Comment:
```
return chainComparator<VendorItem>(compareBy((item) => item.item?.itemCategoryHashes.slice(-1)));
```

Reply:
i think slice() still delivers an array here, so you'd need an additional `[0]`

is the goal to compare by final (most specific) ICH?
as far as i can tell, the existing code was stringifying the arrays and then string comparing in left to right order, so (roughly) from least specific to most specific

Reply:
just fully woke up and did some testing just drop the slice and compare by [0]
```
return chainComparator<VendorItem>(compareBy((item) => item.item?.itemCategoryHashes.[0]));
```

Reply:
In-Game: 
![image](https://user-images.githubusercontent.com/4798491/133892468-b064e160-cbc0-4464-b9e3-d81b06d4539c.png)

Reply:
weeeird.. wouldn't that just compare them all by [Weapon]?

Reply:
i think ultimately that just naively returns them in the order the API data came back

Reply:
![image](https://user-images.githubusercontent.com/4798491/133937014-8557a592-c7bf-409e-a2a3-a936dca5f4fa.png)
so sorting by [0] sorts by `Kinetic`, `Energy`, and `Power`...